chaoscenter/graphql/server/pkg/chaos_infrastructure/service.go

```diff
@@ -86,6 +86,11 @@ func (in *infraService) RegisterInfra(c context.Context, projectID string, input
 		return nil, errors.New("infra already exists in this project with the same name")
 	}
 
+	// Check if EnvironmentID is valid
+	if _, err := in.envOperator.GetEnvironmentDetails(c, input.EnvironmentID, projectID); err != nil {
+		return nil, errors.New("Invalid EnvironmentID")
```

Comment:
@Dhanush0369 have you tested these changes?

Reply:
Yes, I have attached the graphql playground pic